### DISCUSSION TITLE
In update flows, access the pre-update values

### DISCUSSION THREAD

**Discussion by gdpdils** [directus/directus#14489] [Archive]
I don't know if this is already possibile and I've missed it, but setting up a _flow_ triggered by the "update" event on a collection, how do I access the previous value of the row(s) being updated? `{{$trigger}}` gives me the updated value, but I would like to be able to access the value _before_ the update. For instance, in MySql an update trigger provides the pre-set variables `OLD` and `NEW` with such values.

Thanks

**Comment by rijkvanzanten** [comment]
If you use a `filter` trigger, you should be able to read the current item from the database, as the flow will be executed before the item is saved to the database

**Reply by paulleduc** [reply]
Are there any plans to add support for this for "Action (Non-blocking)" event triggers? Otherwise it's not really possible to know if a particular field value has changed for non-blocking triggers. I find most of my desired flows take action only when a particular field has changed.

**Reply by rijkvanzanten** [reply]
Actions fire after the updated item was saved, so at that point, you can't read the pre-saved values anymore (as those are already replaced).

> Otherwise it's not really possible to know if a particular field value has changed for non-blocking triggers

Knowing if a field was changed isn't as difficult, as the payload of the update action is passed in the $trigger payload, so you see the fields that were updated, and what they were updated to. You can't however read what the value was updated from

**Comment by luciandavidescu** [comment]
You can look into directus_revisions, fetch the last but one for that type of item with offset=1 and limit=1
